### DISCUSSION TITLE
RPI5  Can NOT install openhantek6022

### DISCUSSION THREAD

**Discussion by skippy2235** [OpenHantek/OpenHantek6022#403] [General]
I am very new to RPI.  I have a RPI5 and use it alot, I have installed a few things.
I can NOT get any of the releases to work.
My PI uses an aarch64 cpu.  I have tried amd.64, armv7.  nothing I do will work.  Keep getting not recognized and file not found.
Is there any step by step instructions for RPI5 install?
Dave

**Comment by Ho-Ro** [comment]
You could build the SW on the RasPi, this should create the correct binary architecture.

**Reply by skippy2235** [reply]
Do you mean using the source code and compile it on my rpi5?
That is a good idea, but for me a challenge.  
Any tips on where to start learning that?
I am game to learn.
I do alot of arduino, so I'm familiar with idea, never done it.

**Reply by Ho-Ro** [reply]
https://github.com/OpenHantek/OpenHantek6022?tab=readme-ov-file#building-openhantek6022-from-source

**Comment by jonw0224** [comment]
I compiled and run on a Rpi4 and also a Rpi5. The instructions are straightforward and everything works well.JonathanOn May 2, 2024, at 4:08 PM, Martin ***@***.***> wrote:﻿
https://github.com/OpenHantek/OpenHantek6022?tab=readme-ov-file#building-openhantek6022-from-source

**Comment by skippy2235** [comment]
Solved.
Thanks for your help.  Took me 2 hrs
I wrote a HOW to for anyone the wants the easy step by step for the RPI 5
**Tip at bottom of page**

How to install Open Hantek6022 Linux software on the RPI5 with the aarch64 cpu.
There are no releases yet with the aarch64 avail. 
BUT it is supper easy to configure the source code to work.
Just follow these steps.  Should take about 15 min, or less.

1) Download the source code from git hub.  This link downloads code and places it into directly.
When I did this the location was:  /home/pi/OpenHantek6022.
Copy / Paste this line into the Terminal.

git clone https://github.com/OpenHantek/OpenHantek6022.git

2) Since you are using a RPI5 that means you are using the newest version of Bookworm.
So there is not need to worry about GL drivers all that is good to go.
Copy / Paste this line into the Terminal.

apt install g++ make cmake fakeroot qttools5-dev libfftw3-dev binutils-dev libusb-1.0-0-dev libqt5opengl5-dev mesa-common-dev libgl1-mesa-dev libgles2-mesa-dev

3) There is a script that will do most of the magic for us.  It name is LinuxSetup_AsRoot .   To get this file to run we must enter these commands into Terminal.  (we must change dir to where the file is at)
Copy / Paste this lines into the Terminal one line at a time.

chmod +x LinuxSetup_AsRoot
sudo ./LinuxSetup_AsRoot

4)  After you've installed above,  we must enter these commands into Terminal. (we must change dir to where the file is at)
Copy / Paste this lines into the Terminal one line at a time.

mkdir build
cd build
cmake ..
make -j4


5) Now we need to build the files.  To get this file to run we must enter these commands into Terminal. (we must change dir to where the file is at)
Copy / Paste this lines into the Terminal one line at a time.

chmod +x [LinuxBuild](https://github.com/OpenHantek/OpenHantek6022/blob/main/LinuxBuild)
./[LinuxBuild](https://github.com/OpenHantek/OpenHantek6022/blob/main/LinuxBuild)


6) goto the newly created folder called packages and find the _arm64 file.
Copy the entire name, and paste in this line below, behind the sudo install apt
Example:  sudo apt install ./openhantek_3.4-rc3-dirty_arm64.deb
Copy / Paste this lines into the Terminal.

sudo apt install ./openhantek name of your file here arm64.deb



7) Done, 
To Run OpenHantek6022
On a Linux system start the program via the menu entry OpenHantek (Digital Storage Oscilloscope) or from a terminal window as OpenHantek.





Tips:  
It makes it real easy if you setup Root user file Manager.
Menu > Preferences > Main Menu Editor, Accessories, New Item
Name: Root File Manager
Command: sudo pcmanfm
Comment: Open the file manager as root user
Add the item and you will then have the new item appear in the start menu accessories folder.
